src/python/zquantum/qaoa/ansatzes/x_ansatz.py

```diff
@@ -0,0 +1,117 @@
+from zquantum.core.interfaces.ansatz import Ansatz, ansatz_property
+
+from zquantum.core.circuits import Circuit
+from zquantum.core.evolution import time_evolution
+from zquantum.core.openfermion import change_operator_type
+
+from openfermion import QubitOperator, IsingOperator
+from openfermion.utils import count_qubits
+from typing import Tuple, Union, Optional, List
+import numpy as np
+import sympy
+from overrides import overrides
+from itertools import combinations
+
+
+class QAOAXAnsatz(Ansatz):
+
+    supports_parametrized_circuits = True
+    cost_hamiltonian = ansatz_property("cost_hamiltonian")
+
+    def __init__(
+        self,
+        number_of_layers: int,
+        cost_hamiltonian: Union[QubitOperator, IsingOperator],
+    ):
+        """This is implementation of the X Ansatz from https://arxiv.org/abs/2105.01114
+
+        Args:
+            number_of_layers: k-body depth (the maximum number of qubits entangled at one time) as described in https://arxiv.org/abs/2105.01114. Cannot be greater than the number of qubits.
+            cost_hamiltonian: Hamiltonian representing the cost function
+
+        Attributes:
+            number_of_qubits: number of qubits required for the ansatz circuit.
+            number_of_params: number of the parameters that need to be set for the ansatz circuit.
+        """
+
+        super().__init__(number_of_layers)
+        self._cost_hamiltonian = cost_hamiltonian
+
+        assert number_of_layers <= self.number_of_qubits
+
+    @property
+    def number_of_qubits(self):
+        """Returns number of qubits used for the ansatz circuit."""
+        return count_qubits(change_operator_type(self._cost_hamiltonian, QubitOperator))
+
+    @property
+    def number_of_params(self) -> int:
+        """Returns number of parameters in the ansatz."""
+        sum = 0
+        for i in range(1, self.number_of_layers + 1):
+            sum += ncr(self.number_of_qubits, i)
+        return sum
+
+    @overrides
+    def _generate_circuit(self, params: Optional[np.ndarray] = None) -> Circuit:
+        """Returns a parametrizable circuit represention of the ansatz.
+        Args:
+            params: parameters of the circuit.
+        """
+        if params is not None:
+            Warning(
+                "This method retuns a parametrizable circuit, params will be ignored."
+            )
+        circuit = Circuit()
+
+        # Add time evolution layers
+        j = 0
+        for k in range(self.number_of_layers):
+
+            A = combinations(range(0, self.number_of_qubits), k + 1)
+
+            for S in list(A):
+                H_j = QubitOperator(" ".join([f"X{i}" for i in S]))
+
+                circuit += time_evolution(H_j, sympy.Symbol(f"theta_{j}"))
+                j += 1
+
+        return circuit
+
+
+def create_x_qaoa_circuits(
```

Comment:
TBH I don't think this is very important, but since you've already implemented it, it doesn't hurt to keep it.